vic/drivers/cesm/src/get_global_param.c

```diff
@@ -0,0 +1,753 @@
+ * @brief    Validate the filenames_struct
+ *****************************************************************************/
+void
+validate_filenames(filenames_struct *filenames)
+{
+    extern option_struct options;
+
+    // Validate log directory
+    if (strcmp(filenames->result_dir, "MISSING") == 0) {
+        log_err("Log directory must be specified in CESM driver");
+    }
+
+    // Validate result directory
+    if (strcmp(filenames->result_dir, "MISSING") == 0) {
+        log_err("No results directory has been defined.  Make sure that the "
+                "global file defines the result directory on the line that "
+                "begins with \"RESULT_DIR\".");
+    }
+
+    // Validate soil parameter file information
+    if (strcmp(filenames->soil, "MISSING") == 0) {
+        log_err("No soil parameter file has been defined.  Make sure that the "
+                "global file defines the soil parameter file on the line that "
+                "begins with \"SOIL\".");
+    }
+
+    // Validate veg parameter information
+    if (strcmp(filenames->veg, "MISSING") == 0) {
+        log_err("No vegetation parameter file has been defined.  Make sure "
+                "that the global file defines the vegetation parameter "
+                "file on the line that begins with \"VEGPARAM\".");
+    }
+    if (strcmp(filenames->veglib, "MISSING") == 0) {
+        log_err("No vegetation library file has been defined.  Make sure "
+                "that the global file defines the vegetation library file "
+                "on the line that begins with \"VEGLIB\".");
+    }
+
+    // Validate snow parameter file
+    if (options.SNOW_BAND > 1) {
+        log_err("Snowbands not implemented in CESM driver");
+    }
+
+    // Validate lake parameter information
+    if (options.LAKES) {
+        log_err("Lakes are not implemented in CESM driver");
+    }
+}
+
+/******************************************************************************
+ * @brief    Validate the global_param_struct
+ *****************************************************************************/
+void
+validate_global_param(global_param_struct *gp)
+{
+    if (gp->model_steps_per_day != gp->snow_steps_per_day) {
+        log_err("snow_steps_per_day must match model_steps_per_day");
+    }
+    if (gp->model_steps_per_day != gp->runoff_steps_per_day) {
+        log_err("runoff_steps_per_day must match model_steps_per_day");
+    }
+    if (gp->model_steps_per_day != gp->atmos_steps_per_day) {
+        log_err("atmos_steps_per_day must match model_steps_per_day");
+    }
+    if (gp->model_steps_per_day != gp->output_steps_per_day) {
+        log_err("output_steps_per_day must match model_steps_per_day");
+    }
+}
+
+/******************************************************************************
+ * @brief    Validate the option_struct
+ *****************************************************************************/
+void
+validate_options(option_struct *options)
+{
```

Comment:
The structure of this function could be shared, but here we validate specific options that apply only to this driver.